src/main/java/org/killbill/billing/plugin/gocardless/GoCardlessPaymentPluginApi.java

```diff
@@ -257,36 +253,31 @@ public PaymentTransactionInfoPlugin refundPayment(UUID kbAccountId, UUID kbPayme
 
 	public List<PaymentTransactionInfoPlugin> getPaymentInfo(UUID kbAccountId, UUID kbPaymentId,
 			Iterable<PluginProperty> properties, TenantContext context) throws PaymentPluginApiException {
-		
-		logger.info("getPaymentInfo");
+		logger.info("getPaymentInfo, kbAccountId={}", kbAccountId);
 		List<PaymentTransactionInfoPlugin> paymentTransactionInfoPluginList = new ArrayList<>();
 		String mandateId = getMandateId(kbAccountId, context) ;
-		logger.info("Mandate="+mandateId);
 		Mandate mandate = client.mandates().get(mandateId).execute(); //get GoCardless Mandate object
 		String customerId = mandate.getLinks().getCustomer(); //retrieve customer id from mandate
-		logger.info("CustomerId="+customerId);
 		
 		Iterable<Payment> payments = client.payments().all().withCustomer(customerId).execute(); //get all payments related to customer
 		
 		for (Payment payment : payments) {
 			String kbPaymentIdFromPayment = payment.getMetadata().get("kbPaymentId"); //get kbPaymentId from metadata in payment
-			logger.info("kbPaymentIdFromPayment="+kbPaymentIdFromPayment);
 			if(kbPaymentIdFromPayment != null && kbPaymentId.toString().equals(kbPaymentIdFromPayment)) {
-				logger.info("Found matching payment");
 				Currency killBillCurrency = convertGoCardlessCurrencyToKillBillCurrency(payment.getCurrency());
-				logger.info("payment_status="+payment.getStatus());
 				PaymentPluginStatus status = convertGoCardlessToKillBillStatus(payment.getStatus());
 				String kbTransactionPaymentIdStr = payment.getMetadata().get("kbTransactionId"); 
 				UUID kbTransactionPaymentId = kbTransactionPaymentIdStr !=null?UUID.fromString(kbTransactionPaymentIdStr):null;
-				logger.info("kbTransactionPaymentId="+kbTransactionPaymentId);
 				List<PluginProperty> outputProperties = new ArrayList<PluginProperty>();
 				outputProperties.add(new PluginProperty("mandateId",mandateId,false)); //arbitrary data to be returned to the caller
 				outputProperties.add(new PluginProperty("customerId",customerId,false));  //arbitrary data to be returned to the caller
 				outputProperties.add(new PluginProperty("gocardlessstatus",payment.getStatus(),false)); //arbitrary data to be returned to the caller
 				GoCardlessPaymentTransactionInfoPlugin paymentTransactionInfoPlugin = new GoCardlessPaymentTransactionInfoPlugin(
-						kbPaymentId, kbTransactionPaymentId, TransactionType.PURCHASE, new BigDecimal(payment.getAmount()), killBillCurrency,
+						kbPaymentId, kbTransactionPaymentId, TransactionType.PURCHASE, // In a real-world plugin, set to the appropriate TransactionType
+						new BigDecimal(payment.getAmount()), killBillCurrency, //TODO, There is a bug here, payment.getAmount() returns the amount in minor units, this needs to be converted
```

Comment:
Check https://github.com/killbill/killbill-adyen-plugin/blob/5bb6ad36a868bab15a4123af32a8c453f383e3b7/src/main/java/org/killbill/billing/plugin/adyen/client/model/NotificationItem.java#L53, that should do it?

We could also add a helper `fromMinorUnits` here: https://github.com/killbill/killbill-plugin-framework-java/blob/20ae3d09d13d9f6188be79f21694ae59fd23a104/src/main/java/org/killbill/billing/plugin/util/KillBillMoney.java#L28